Iter23

havilcorp opened the pull request.

Commit:
iter21

Commit:
iter22

Commit:
iter23

tmvrus commented on internal/config/agent/agent.go:
```diff
@@ -0,0 +1,135 @@
+
+	if envPoolInterval := os.Getenv("POLL_INTERVAL"); envPoolInterval != "" {
+		envPoolIntervalVal, err := strconv.Atoi(envPoolInterval)
+		if err != nil {
+			return nil
```
?

tmvrus commented on internal/config/server/server_test.go:
```diff
@@ -0,0 +1,21 @@
+
+func TestConfig_WriteServerConfig(t *testing.T) {
+	conf := NewServerConfig()
+	t.Run("Test server", func(t *testing.T) {
```
делать только один вложенный тест довольно странно

tmvrus commented on internal/middleware/crypto.go:
```diff
@@ -46,3 +47,32 @@ func HashSHA256Middleware(key string) func(http.Handler) http.Handler {
 	}
 }
+
+func RsaMiddleware(filepath string) func(http.Handler) http.Handler {
```
по-хорошему, Rsa->RSA потому что это аббревиатура

tmvrus commented on internal/middleware/crypto.go:
```diff
@@ -46,3 +47,32 @@ func HashSHA256Middleware(key string) func(http.Handler) http.Handler {
+func RsaMiddleware(filepath string) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			body, err := io.ReadAll(r.Body)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+			err = r.Body.Close()
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+			priv, err := cryptorsa.LoadPrivateKey(filepath)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+			bodyDecode, err := cryptorsa.DecryptOAEP(priv, body)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
```
тут никак не используется ошибка, а можно было ее логировать или показывать пользователю, что бы он понял в чем ошибка

tmvrus commented on internal/middleware/crypto_test.go:
```diff
@@ -38,3 +39,32 @@ func TestHashSHA256Middleware(t *testing.T) {
+		if err != nil {
+			t.Error(err)
+		}
+		w.WriteHeader(200)
```
200 -> http.StatusOK

tmvrus commented on internal/repositories/metric/storage/file/file_test.go:
```diff
@@ -5,12 +5,24 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// func TestNewFileStorage(t *testing.T) {
```
закоментированного кода в рабочей версии быть не должно, это дурной тон

tmvrus commented on internal/repositories/metric/storage/file/file_test.go:
```diff
@@ -97,4 +109,13 @@ func TestFileStorage_SaveToFile_LoadFromFile(t *testing.T) {
 	if err != nil {
 		t.Errorf("SaveToFile %v", err)
 	}
+	conf2 := server.Config{
```
https://habr.com/ru/companies/dododev/articles/714512/

tmvrus reviewed (changes requested):
есть мелкие замечания

Commit:
- поправил вложенность в тестах
- написал простенький анализатор для проверки заккоментированного кода
- иные правки
